packages/@storefront/structure/test/unit/generic-paging.ts

```diff
@@ -0,0 +1,284 @@
+import GenericPaging from '../../src/generic-paging';
+import suite from './_suite';
+
+suite('GenericPaging', ({ expect, spy, stub, itShouldProvideAlias }) => {
+  let genericPaging: GenericPaging;
+
+  beforeEach(() => {
+    genericPaging = new GenericPaging();
+  });
+
+  itShouldProvideAlias(GenericPaging, 'paging');
+
+  describe('constructor()', () => {
+    describe('props', () => {
+      it('should have initial value', () => {
+        expect(genericPaging.props).to.eql({
+          showIcons: true,
+          showLabels: true,
+          numericLabels: false,
+          labels: { first: 'First', next: 'Next', prev: 'Prev', last: 'Last' },
+          icons: {},
+          pageSize: 20,
+          current: 1,
+          itemCount: 0,
+          limit: 5,
+        });
+      });
+    });
+
+    describe('state', () => {
+      it('should have initial values', () => {
+        expect(genericPaging.state).to.deep.include({
+          range: [],
+          lowOverflow: false,
+          highOverflow: false
+        });
+      });
+
+      function testHandler(handlerName: string, propName: string) {
+        describe(handlerName + '()', () => {
+          it('should not throw when no props are passed in', () => {
+            expect(() => genericPaging.state[handlerName]()).to.not.throw();
+          });
+
+          it('should wrap the function passed in as props', () => {
+            const handler = spy();
+            genericPaging.props = { [propName]: handler };
+
+            genericPaging.state[handlerName]();
+
+            expect(handler).to.be.called;
+          });
+        });
+      }
+
+      testHandler('firstPage', 'onFirstPage');
+      testHandler('lastPage', 'onLastPage');
+      testHandler('prevPage', 'onPrevPage');
+      testHandler('nextPage', 'onNextPage');
+
+      describe('switchPage()', () => {
+        it('should not throw when no props are passed in', () => {
+          expect(() => genericPaging.state.switchPage(5)()).to.not.throw();
+        });
+
+        it('should wrap the function passed in as props', () => {
+          const onSwitchPage = spy();
+          const page = 5;
+          genericPaging.props = { onSwitchPage };
+
+          genericPaging.state.switchPage(page)();
+
+          expect(onSwitchPage).to.be.calledWith(page);
+        });
+      });
+    });
+  });
+
+  describe('init()', () => {
+    it('should call updateState()', () => {
+      const updateState = genericPaging.updateState = spy();
+
+      genericPaging.init();
+
+      expect(updateState).to.be.called;
+    });
+  });
+
+  describe('onUpdate()', () => {
+    it('should call updateRange()', () => {
```

Comment:
Name and test mismatch: case description references `updateRange`, but body tests `updateState`.

Reply:
Will do.